src/core/templates/show-logo.js

```diff
@@ -1,35 +1,45 @@
 // @ts-check
+import { docLink, showError } from "../../core/utils.js";
 import { html } from "../../core/import-maps.js";
-import { showWarning } from "../../core/utils.js";
 
 const name = "core/templates/show-logo";
 
 /**
+ * Logo mapper. Takes a logo structure and converts it to HTML.
+ *
  * @param {object} logo
  * @param {string} logo.src
  * @param {string} logo.url
  * @param {string} logo.alt
  * @param {string} [logo.id]
  * @param {number} [logo.width]
  * @param {number} [logo.height]
+ * @param {number} index
  */
-export default function showLogo(logo) {
+export default function showLogo(logo, index) {
   /** @type {HTMLAnchorElement} */
-  const a = html`<a href="${logo.url || ""}" class="logo"></a>`;
-  if (!logo.alt) {
-    const msg = "Found spec logo without an `alt` attribute.";
-    showWarning(msg, name, { elements: [a] });
+  const a = html`<a href="${logo.url ?? null}" class="logo"
+    ><img
+      id="${logo.id ?? null}"
+      alt="${logo.alt ?? null}"
+      width="${logo.width ?? null}"
+      height="${logo.height ?? null}"
+      crossorigin
+    />
+  </a>`;
+  const img = a.querySelector("img");
+  if (!img.alt) {
+    const msg = `Logo at index ${index} is missing required "\`alt\`" property.`;
+    const hint = docLink`Add the missing "\`alt\`" property describing the logo. See ${"[logos]"} for more information.`;
+    showError(msg, name, { hint, elements: [img] });
+  }
+  // avoid triggering 404 requests from dynamically generated hyperHTML attribute values
```

Comment:
Is this still required? What was the original issue here?

Reply:
Remember a few years ago it was messing up our tests by creating 404s. When the image is created with a src in the markup, it would always trigger a download.

Reply:
Good news! Seems this got fixed!